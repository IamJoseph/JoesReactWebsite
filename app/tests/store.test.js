/**
 * Test store addons
 */

import { browserHistory } from 'react-router';
import toJson from 'enzyme-to-json';
import configureStore from '../store';

describe('configureStore', () => {
  let store;

  it('returns a store object', () => {
    store = configureStore();
    expect(typeof store).toBe('object');
    expect(toJson(store)).toMatchSnapshot();
  });

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  describe('asyncReducers', () => {
    it('should contain an object for async reducers', () => {
      expect(typeof store.asyncReducers).toBe('object');
    });
  });

  describe('runSaga', () => {
    it('should contain a hook for `sagaMiddleware.run`', () => {
      expect(typeof store.runSaga).toBe('function');
    });
  });
});
