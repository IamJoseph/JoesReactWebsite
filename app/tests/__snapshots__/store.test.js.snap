// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configureStore returns a store object 1`] = `undefined`;
